public/pages/Main/Main.tsx

```diff
@@ -329,495 +355,548 @@ export default class Main extends Component<MainProps, MainState> {
-                                ROUTES.ALIASES,
-                                ROUTES.DATA_STREAMS,
-                                ROUTES.TEMPLATES,
-                                ROUTES.CREATE_DATA_STREAM,
-                                ROUTES.CREATE_TEMPLATE,
-                                ROUTES.COMPOSABLE_TEMPLATES,
-                                ROUTES.CREATE_COMPOSABLE_TEMPLATE,
-                              ]}
-                              render={() => (
-                                <DataSourceMenu
-                                  appName={"Index State Management"}
-                                  setMenuMountPoint={this.props.setActionMenu}
-                                  showDataSourceSelectable={true}
-                                  dataSourceCallBackFunc={({ id: dataSourceId, label: dataSourceLabel }) => {
-                                    this.setState({ dataSourceId, dataSourceLabel });
-                                  }}
-                                  disableDataSourceSelectable={false}
-                                  notifications={services.notificationService}
-                                  savedObjects={core.savedObjects.client}
-                                  selectedOption={(() => {
-                                    if (this.state.dataSourceId && this.state.dataSourceId !== "") {
-                                      return [
-                                        {
-                                          id: this.state.dataSourceId,
-                                          label: this.state.dataSourceLabel,
-                                        },
-                                      ];
-                                    }
-                                    return undefined;
-                                  })()}
-                                  fullWidth={false}
-                                  hideLocalCluster={false}
-                                />
-                              )}
-                            />
-                          </Switch>
-                        )}
-                        <ModalRoot services={services} />
-                        <EuiPage restrictWidth="100%">
-                          {/* Hide side navigation bar when creating or editing rollup job*/}
-                          {!HIDDEN_NAV_ROUTES.includes(pathname) &&
-                          !HIDDEN_NAV_STARTS_WITH_ROUTE.some((item) => pathname.startsWith(item)) ? (
-                            <EuiPageSideBar style={{ minWidth: 200 }}>
-                              <EuiSideNav style={{ width: 200 }} items={sideNav} />
-                            </EuiPageSideBar>
-                          ) : null}
-                          <EuiPageBody>
+                        <DataSourceMenuReadOnlyContext.Provider
+                          value={{
+                            dataSourceReadOnly: this.state.dataSourceReadOnly,
```

Comment:
why can't we have this functions as part of DataSourceMenuContext as both are related instead of having a separate context ?

Reply:
To take a step back, why would we need to put this properties in context, as there are only two properties, we can send in props for the two pages.

Reply:
Passed as props instead of context